src/Parser/IncludeProcessor/DefaultIncludeProcessor.php

```diff
@@ -75,6 +76,42 @@ public function process(ParserInterface $parser, string $file, array $data): arr
 
         $this->included[$file] = true;
 
+        // The order the the include statements needs to be reversed. Indeed when merging two sets of data, e.g.
```

Comment:
That change of order needs to be documented. The current behaviour in 3.x is in line with 2.x but I think it's a bug and doesn't make sense from a user PoV. Bug that would be a BC break still (which is worth fixing IMO since it's a bug)